docs/drafts/derive-serviceaccount.md

```diff
@@ -0,0 +1,394 @@
+
+The ServiceAccount is specified in the ClusterExtension manifest as shown below:
+
+```yaml
+apiVersion: olm.operatorframework.io/v1alpha1
+kind: ClusterExtension
+metadata:
+  name: argocd
+spec:
+  source:
+    sourceType: Catalog
+    catalog:
+      packageName: argocd-operator
+      version: 0.6.0
+  install:
+    namespace: argocd
+    serviceAccount:
+      name: argocd-installer
+```
+
+The initial stable version (v1.0.0) only supports FBC catalogs containing registry+v1 bundles. OLMv1 will not support all OLMv0 content. OLMv1 will only support bundles that meet the following criteria:
+* AllNamespaces install mode is enabled
+* No dependencies on other packages or GVKs
+* No webhooks
+* Does not make use of the OperatorConditions API
+
+### Required RBAC
+
+You can determine the specifics of these permissions by referencing the bundle of the ClusterExtension you want to install. The service account must have the following permissions:
+
+* ClusterRole with all the roles specified in the bundle ClusterServiceVersion. This includes all the
+  rules defined in the CSV under `.spec.install.clusterPermissions` and `.spec.install.permissions`
+* ClusterRole to create and manage CustomResourceDefinitions
+* Update finalizers on the ClusterExtension to be able to set blockOwnerDeletion and ownerReferences
+* Permissions to create the controller deployment, this corresponds to the rules to manage the
+  deployment defined in the ClusterServiceVersion
+* Permissions to create the other manifest objects, rules to manage any other cluster-scoped resources
+  shipped with the bundle
+* Rules to manage any other namespace-scoped resources 
+* Permissions to create the necessary roles and rolebindings for the controller to be able to perform its job
+* Get, list, watch, update, patch, delete the specific resources that get created/
+
+### Manual process for minimal RBAC creation
+
+There are no production tools available to help you understand the precise RBAC required for your cluster extensions. However, if you want to figure this manually you can try the below:
+
+* Create all the intial rbac and then iterate over the ClusterExtention failures, examining conditions and updating the RBAC to include the generated cluster role names (name will be in the failure condition).
+* After reading the failure condition, update the installer RBAC and iterate until you are out of errors.
+* You can get the bundle image, unpacking the same and inspect the manifests to figure out the required permissions.
+* The `oc` cli-tool creates cluster roles with a hash in their name, you canquery the newly created ClusterRole names, reduce the installer RBAC scope to have the ClusterRoles needed ,this can include generated roles. You can achieve this by allowing the installer to get, list, watch and update any cluster roles.
```

Comment:
IMO, we should decide on one of these approaches for now and document the end-to-end flow a user would take to do it.

Reply:
discuss how we can do an e2e example

Reply:
added an e2e example